gradle-plugin/src/test/resources/javaOutput/search-micronaut/SearchResource.java.txt

```diff
@@ -0,0 +1,51 @@
+package org.acme.rest.v1.search;
+
+import io.micronaut.core.convert.format.Format;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Options;
+import io.micronaut.http.annotation.Produces;
+import io.micronaut.http.annotation.QueryValue;
+import io.micronaut.validation.Validated;
+import java.time.LocalDate;
+import javax.annotation.Generated;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.validation.Valid;
+
+/**
+ * This resource generates the search endpoint
+ */
+@Generated(
+    comments = "Specification filename: search.v1.json",
+    value = "ch.silviowangler.restapi"
+)
+@Controller("v1/search")
+@Validated
+public class SearchResource {
+  public static final String OPTIONS_CONTENT = "{\"general\":{\"name\":\"search.v1\",\"description\":\"This resource generates the search endpoint\",\"version\":\"1.0.0\",\"x-route\":\"v1/search\",\"lifecycle\":{\"deprecated\":false,\"info\":\"This version is still valid\"}},\"verbs\":[{\"verb\":\"GET_ENTITY\",\"rel\":\"entity\",\"responseStates\":[{\"code\":200,\"message\":\"200 Ok\",\"comment\":\"content in response body\"},{\"code\":503,\"message\":\"503 Service Unavailable\",\"comment\":\"Backend server not reachable or too slow\"}],\"representations\":[{\"name\":\"json\",\"comment\":\"\",\"responseExample\":\"{...}\",\"isDefault\":true,\"mimetype\":\"application/json\"}],\"parameters\":[{\"name\":\"q\",\"type\":\"string\",\"options\":null,\"mandatory\":true,\"min\":null,\"max\":null,\"multiple\":false,\"defaultValue\":null,\"alias\":[],\"x-comment\":\"Search\"},{\"name\":\"startDate\",\"type\":\"date\",\"options\":null,\"mandatory\":false,\"min\":null,\"max\":null,\"multiple\":false,\"defaultValue\":null,\"alias\":[],\"x-comment\":\"start date\"}],\"permissions\":[]}],\"fields\":[{\"name\":\"id\",\"type\":\"uuid\",\"options\":null,\"mandatory\":[],\"min\":null,\"max\":null,\"multiple\":false,\"defaultValue\":null,\"protected\":[],\"visible\":true,\"sortable\":false,\"readonly\":false,\"filterable\":false,\"alias\":[],\"x-comment\":\"id\"}],\"pipes\":[],\"types\":[],\"validators\":[]}";
+
+  private final SearchResourceDelegate delegate;
+
+  @Inject
+  public SearchResource(SearchResourceDelegate delegate) {
+    this.delegate = delegate;
+  }
+
+  @Options(
+      uri = "/.json"
+  )
+  @Produces("application/json")
+  public String getOptions() {
+    return OPTIONS_CONTENT;
+  }
+
+  @Produces("application/json")
+  @Get(
+      uri = "/.json"
+  )
+  public SearchGetResourceModel getEntity(@Valid @QueryValue String q,
+      @Valid @QueryValue @Format("yyyy-MM-dd") @Nullable LocalDate startDate) {
```

Comment:
I am using a HTTP request that has the following query string `?startDate=1978-11-01&endTime=2019-11-01T14:00:00Z`. This does not seem to work since it does not bind the value to an `java.time.Instant`.

Using a specific format does not seem flexible enough. It think we will have to write our own converter there. I release a new version of the plugin 1.4.3 that does the job for you.

- `ch.silviowangler.rest.micronaut.binding.DateFormat` does the conversion to `LocalDate`
- `ch.silviowangler.rest.micronaut.binding.DateTimeFormat` does the conversion to `Instant`

Can you adjust the code generation in order we get code like 

 ```
@Get(
    uri = "/{id}"
  )
  @Produces("application/json")
  public InsurersGetResourceModel getEntity(String id,
                                            @Valid @QueryValue @DateFormat @Nullable LocalDate startDate,
                                            @Valid @QueryValue @DateTimeFormat @NotNull Instant endTime) {

    System.out.println("startDate " + startDate);
    System.out.println("endTime " + endTime);

    return delegate.getEntity(id);
  }
```